### PR TITLE
Convert Image and Ratio to @bolt/element (LitElement)

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
TBD

## Summary
Converts the Image and Ratio components over to LitElement via our new base Class provided by `@bolt/element`, `BoltElement`.

Early testing shows that this should help considerably with JavaScript execution times / client-side rendering performance!

1C219D1C-98B7-45C8-885C-0BC846C22D67
FA9CD71E-393C-4BC0-BB95-935CF32D093B
684D1E36-4C8F-4DCE-8839-92DEF3AB274A

 
https://www.webpagetest.org/video/view.php?id=191112_9a75643968a5f66c62c7dd1a396b3038716cad68

## How to test
- Review changes
- Confirm Travis tests all passing as expected 
- Double-check browsers to confirm everything is working as expected (Edge, IE 11, etc)

### CODE REVIEW

File: packages/components/bolt-button/src/button.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
-import { unsafeCSS } from 'lit-element';
-import { render, html } from 'lit-html';
-import { BoltActionElement } from '@bolt/element';
+import {
+  BoltActionElement,
+  unsafeCSS,
+  render,
+  html,
+  ifDefined,
+} from '@bolt/element';
```
Comment (adamszalapski) [inline packages/components/bolt-button/src/button.js]: I love that we have one central file with all elements needed to import to render Bolt Component

Comment (sghoweri) [reply packages/components/bolt-button/src/button.js]: Me too — that's the whole point! 😁

**Comment by sghoweri** [comment]
> Image is looking snappier than ever. Nice work!
> 
> I did notice a regression on full-bleed background images. Maybe it's ratio related?
> 
> Run locally and compare:
> 
> * Background fill: [local](http://localhost:3000/pattern-lab/patterns/02-components-background-25-background-fill-variations/02-components-background-25-background-fill-variations.html) | [master](https://master.boltdesignsystem.com/pattern-lab/patterns/02-components-background-25-background-fill-variations/02-components-background-25-background-fill-variations.html)
> * Band with background: [local](http://localhost:3000/pattern-lab/patterns/02-components-band-35-band-with-background/02-components-band-35-band-with-background.html) | [master](https://master.boltdesignsystem.com/pattern-lab/patterns/02-components-band-35-band-with-background/02-components-band-35-band-with-background.html)
> 
> In Chrome the vertical centering is different. In IE11 it looks like the images aren't scaled down to fit at all. You just see a small piece of the image, even more so than you used to.
> 
> Also, SVGs aren't lazy loading in IE11:
> 
> * [Lazyload variation](http://localhost:3000/pattern-lab/patterns/02-components-image-20-image-lazyload-variations/02-components-image-20-image-lazyload-variations.html)
> * [Blockquote](http://localhost:3000/pattern-lab/patterns/02-components-blockquote-35-blockquote-theme-variation/02-components-blockquote-35-blockquote-theme-variation.html)
> 
> The SVG bug is happening on master, too. So it's not caused by this PR but it's probably important to fix ASAP.

@danielamorse with the latest updates I just pushed up, this should be all set!

[This update](https://github.com/boltdesignsystem/bolt/pull/1540/files#diff-9d632ea9e7c654cc2b08adafc542ecf3R72) addresses the background image issue. And [this other update](https://github.com/boltdesignsystem/bolt/pull/1540/commits/928d1ad4f3843d4c3ad3498794e2b28506963d4b) addresses the SVG issue (which, fwiw, we can always backport and release separately if necessary).

Should be good to give another look over when you have a chance!

File: packages/components/bolt-image/src/image.js [hunk 2]
```diff
@@ -14,6 +14,10 @@ import '@bolt/core/utils/optimized-resize';
 let cx = classNames.bind(imageStyles);
 let passiveIfSupported = false;
 
+// from https://github.com/aFarkas/lazysizes/blob/gh-pages/plugins/respimg/ls.respimg.js#L24
+const img = document.createElement('img');
+const supportsSrcset = 'sizes' in img && 'srcset' in img;
```
Comment (danielamorse) [inline packages/components/bolt-image/src/image.js]: Clever

File: packages/components/bolt-image/src/image.js [hunk 3]
```diff
@@ -32,49 +33,70 @@ try {
...
+        type: String,
+        attribute: 'placeholder-image',
+      },
+      zoom: {
+        type: Boolean,
+        reflect: true,
+      },
+      cover: {
+        type: Boolean,
+        reflect: true, // fix for bg images not getting the right classes w/ just type: Boolean
```
Comment (danielamorse) [inline packages/components/bolt-image/src/image.js]: Interesting, I read up on reflect [here](https://lit-element.polymer-project.org/api/interfaces/_lib_updating_element_.propertydeclaration.html#reflect). Works for me!

Sounds like combining `reflect` and `converter` gives you some options for handling props that we didn't have before. Very cool!

**Pull request merged**